### PR TITLE
Prisma module 

### PR DESCRIPTION
**Pull request by Kasshern**
# Description

Prisma injection was encapsulated within its own module within the share folder and injected into storage agents like all other provider to avoid repeated instantiation of the postgresql pool

## Related Issue

No issue exists for this. This is preemptive work to including a DB context for prisma txs.

## Motivation and Context

Currently the service instantiates the pool 10+ times

image

## How Has This Been Tested

prisma now logs info upon instantiation of the pool

## Screenshots (if appropriate)
Now only instantiates twice
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/shared/prisma/prisma.service.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ import { PrismaClient } from '@prisma/client';
 
 @Injectable()
 export class PrismaService extends PrismaClient implements OnModuleInit {
+  constructor() {
+    super({ log: ['info'] });
```
Comment (ognjenkurtic) [inline examples/bri-3/src/shared/prisma/prisma.service.ts:7]: why is this needed? default is not info?

**Comment by ognjenkurtic** [comment]
Why is postgresql pool instantiated 2 times?

**Review by ognjenkurtic — APPROVED** [review]

File: examples/bri-3/src/shared/prisma/prisma.service.ts (hunk 1 above)
Comment (skosito) [reply examples/bri-3/src/shared/prisma/prisma.service.ts:7]: yes probably not needed
do we need to call super constructor at all?

File: examples/bri-3/src/bri/communication/api/messages.controller.spec.ts [hunk 2]
```diff
@@ -108,6 +109,8 @@ describe('MessageController', () => {
       .useValue(mockDeep<BpiMessageStorageAgent>())
       .overrideProvider(BpiSubjectStorageAgent)
       .useValue(mockDeep<BpiSubjectStorageAgent>())
+      .overrideProvider(PrismaService)
+      .useValue(mockDeep<PrismaClient>())
```
Comment (skosito) [inline examples/bri-3/src/bri/communication/api/messages.controller.spec.ts:113]: i think this is ok, but what if we add something else to prismaService, wouldn't it make sense to mock PrismaService and not PrismaClient?

**Review by skosito — APPROVED** [review]

**Comment by skosito** [comment]
> Why is postgresql pool instantiated 2 times?

maybe calling super() in prisma service constructor and also connecting onModuleInit?

it looks like we dont need onModuleInit at all: https://docs.nestjs.com/recipes/prisma#use-prisma-client-in-your-nestjs-services

File: examples/bri-3/src/shared/prisma/prisma.service.ts (hunk 1 above)
Comment (Kasshern) [reply examples/bri-3/src/shared/prisma/prisma.service.ts:7]: This is how I got it to log when the pool was instantiated. Can leave or take it at this point. Will leave for now.

File: examples/bri-3/src/bri/communication/api/messages.controller.spec.ts (hunk 2 above)
Comment (Kasshern) [reply examples/bri-3/src/bri/communication/api/messages.controller.spec.ts:113]: Yeah that seems to make the most sense. I will leave that to be changed just-in-time given its a small change.

**Comment by Kasshern** [comment]
> > Why is postgresql pool instantiated 2 times?
> 
> maybe calling super() in prisma service constructor and also connecting onModuleInit?
> 
> it looks like we dont need onModuleInit at all: https://docs.nestjs.com/recipes/prisma#use-prisma-client-in-your-nestjs-services

Will troubleshoot this. If it is the cause will alter the PR and ask for review again.

**Comment by Kasshern** [comment]
> > > Why is postgresql pool instantiated 2 times?
> > 
> > 
> > maybe calling super() in prisma service constructor and also connecting onModuleInit?
> > it looks like we dont need onModuleInit at all: https://docs.nestjs.com/recipes/prisma#use-prisma-client-in-your-nestjs-services
> 
> Will troubleshoot this. If it is the cause will alter the PR and ask for review again.

@ognjenkurtic @skosito  removed onModuleInit. Now initis lazily. Left logging: info as we develop into the future. Can remove at any time. Please review again.

**Review by Therecanbeonlyone1969 — DISMISSED** [review]
LGTM

**Review by skosito — DISMISSED** [review]

**Comment by Therecanbeonlyone1969** [comment]
@Kasshern could you resolve the conflicts to be able to merge

**Comment by Kasshern** [comment]
> @Kasshern could you resolve the conflicts to be able to merge

@Therecanbeonlyone1969 @skosito @Ybittan conflicts resolved now and prisma service injected into migrated bpiAccountsStorage agent. Please approve again.

**Review by Ybittan — APPROVED** [review]
LGTM

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM
